src/find-menu.js

```diff
@@ -26,42 +33,171 @@ class FindMenu extends HTMLElement {
     await this.loadSVG(this.nextButton, 'peersky://static/assets/svg/down.svg');
     await this.loadSVG(this.hideButton, 'peersky://static/assets/svg/close.svg');
 
+    // Setup foundInPage listener on webview
+    this.setupFoundInPageListener();
+    
+    // Setup webview navigation events to detect PDFs
+    this.setupWebviewNavigationListener();
+
     this.input.addEventListener('input', (e) => {
       const { value } = this;
-      if (!value) return;
-      if (value.length > 0 && value !== this.currentSearchValue) {
-        this.currentSearchValue = value;
-        this.resetSearch();
+      if (!value) {
+        this.stopFindInPage('clearSelection');
+        return;
       }
-      this.dispatchEvent(new CustomEvent('next', { detail: { value } }));
+      
+      this.findInWebview(value, { forward: true });
     });
 
     this.input.addEventListener('keydown', ({ keyCode, shiftKey }) => {
       if (keyCode === 13) {
         const { value } = this;
         if (!value) return this.hide();
-        if (value.length > 0 && value !== this.currentSearchValue) {
-          this.currentSearchValue = value;
-          this.resetSearch();
-        }
-        const direction = shiftKey ? 'previous' : 'next';
-        this.dispatchEvent(new CustomEvent(direction, { detail: { value, findNext: true } }));
+        
+        const forward = !shiftKey;
+        this.findInWebview(value, { forward, findNext: true });
       }
     });
 
     this.previousButton.addEventListener('click', () => {
       const { value } = this;
       if (!value) return;
-      this.dispatchEvent(new CustomEvent('previous', { detail: { value, findNext: false } }));
+      this.findInWebview(value, { forward: false, findNext: true });
     });
+    
     this.nextButton.addEventListener('click', () => {
       const { value } = this;
       if (!value) return;
-      this.dispatchEvent(new CustomEvent('next', { detail: { value, findNext: true } }));
+      this.findInWebview(value, { forward: true, findNext: true });
     });
+    
     this.hideButton.addEventListener('click', () => this.hide());
   }
 
+  setupWebviewNavigationListener() {
+    const webview = this.getWebviewElement();
+    if (!webview) return;
+
+    // Listen for did-navigate events to detect content type
+    webview.addEventListener('did-navigate', () => {
+      this.detectContentType();
+    });
+    
+    webview.addEventListener('did-navigate-in-page', () => {
+      this.detectContentType();
+    });
+    
+    // Also check when loading finishes
+    webview.addEventListener('did-finish-load', () => {
+      this.detectContentType();
+    });
+  }
+
+  async detectContentType() {
+    const webview = this.getWebviewElement();
```

Comment:
```suggestion
async detectContentType() {
    if (this.isPdf !== null) return; // Cache it
    const webview = this.getWebviewElement();
```

`detectContentType()` executes on every navigation event, even when `isPdf` is already determined. Adding this caching check prevents redundant execution, optimizing performance.